packages/@lwc/engine-core/src/framework/base-bridge-element.ts

```diff
@@ -121,9 +142,25 @@ export function HTMLBridgeElementFactory(
     const { attributeChangedCallback: superAttributeChangedCallback } = SuperClass.prototype as any;
     const { observedAttributes: superObservedAttributes = [] } = SuperClass as any;
     const descriptors: PropertyDescriptorMap = create(null);
+
+    // present a hint message so that developers are aware that they have not decorated property with @api
+    if (process.env.NODE_ENV !== 'production') {
+        for (const propName in protoDescriptors) {
+            const propDescriptor = protoDescriptors[propName];
+            if (
+                (propDescriptor.get || propDescriptor.set) &&
+                publicProperties.indexOf(propName) === -1 &&
+                trackFields.indexOf(propName) === -1
```

Comment:
You can use `ArrayIndexOf` from `@lwc/shared`.

Reply:
What is the difference between using this method and the one I was using?